data_structures/linked_list/doubly_linked_list.c

```diff
@@ -0,0 +1,259 @@
+/**
+ * @file
+ * @brief   Doubly linked list, simple code.
+ * @details
+ * A doubly linked list is a data structure with a sequence
+ * of components called nodes. Within that nodes there are
+ * three elements: a value recorded, a pointer to the next
+ * node, and a pointer to the previous node.
+ *
+ * In this implementation, the functions of creating the list,
+ * inserting by position, deleting by position, searching 
+ * for value, printing the list, and an example of how the
+ * list works were coded.
+ *
+ * @author  [Gabriel Mota Bromonschenkel Lima](https://github.com/GabrielMotaBLima)
+ */
+#include <stdio.h>
+#include <stdlib.h>
+
+/**
+ * @brief Doubly linked list struct
+ */
+typedef struct list
+{
+    double value;                ///< value saved on each node
+    struct list *next, *prev;  ///< directing to other nodes or NULL
+} List;
+
+/**
+ * @brief   Create list function, a new list containing one node will be created
+ * @param   value       a value to be saved into the first list node
+ * @returns new_list    the list created
+ */
+List *create(double value);
+
+/**
+ * @brief   Insertion by position into the list function
+ * @param   list    a doubly linked List
+ * @param   value   a value to be inserted into the list
+ * @param   pos     a position into the list for value insertion
+ * @returns  list   the input list with a node more or the same list
+ */
+List *insert(List *list, double value, int pos);
+
+/**
+ * @brief   Deletion by position into the list function
+ * @param   list    a doubly linked List
+ * @param   pos     a position into the list for value Deletion
+ * @returns list    the input list with deleted values or the same list
+ */
+List *delete(List *list, int pos);
+
+/**
+ * @brief   Search value into the list function
+ * @param   list    a doubly linked list
+ * @param   value   a value to be looked for into the list
+ * @returns `1`     if the looked up value exists
+ * @returns `0`     if the looked up value doesn't exist
+ */
+int search(List *list, double value);
+
+/**
+ * @brief   Print list function
+ * @param   list    a doubly linked List
+ * @returns void
+ */
+void print(List *list);
+
+/**
+ * @brief   Example function
+ * @returns void
+ */
+void example();
+
+/**
+ * @brief   Main function
+ * @returns 0   on exit
+ */
+int main()
+{
+    // examples for better understanding
+    example();
+    // code here
+    return 0;
+}
+
+List *create(double value)
```

Comment:
This function (and other functions below) are missing function documentation.

Reply:
I documented their header. Is it necessary to do the same with them? :thinking:

Reply:
I think it's better to do so; I don't exactly if the Doxygen is
going to generate them well without documenting them here.

Reply:
Understood, then I will change it in a new commit.

Reply:
Could it be the same comment as the headers?

Reply:
> Could it be the same comment as the headers?

Sure, sounds good. 👍

Reply:
Fixed up, thank you! :slightly_smiling_face: